pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// APIResourceEnablement is the interface to determine which versions and resources are enabled
+type APIResourceEnablement interface {
```

Comment:
OK. In that case, how about "APIResourceConfigSource" for the name of the
interface?

On Tue, Mar 22, 2016 at 1:31 PM, David Eads <notifications@github.com>
wrote:

> In pkg/genericapiserver/enablement_config.go
> <https://github.com/kubernetes/kubernetes/pull/23067#discussion_r57063206>
> :
>
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package genericapiserver
> > +
> > +import (
> > +	"k8s.io/kubernetes/pkg/api/unversioned"
> > +	"k8s.io/kubernetes/pkg/util/sets"
> > +)
> > +
> > +// APIResourceEnablement is the interface to determine which versions and resources are enabled
> > +type APIResourceEnablement interface {
>
> Yeah. I'm just saying that consumers can define the interface, it doesn't
> necessarily have to be defined here. (unless genericapiserver itself is a
> consumer)
>
> Yeah, genericapiserver.Config is the primary consumer. It uses this for
> determining which resources are enabled to make storage decisions.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23067/files/cdef546dee0c2d222b6ca0cfee281079b5be2ce6#r57063206>
>